### DISCUSSION TITLE
Mismatch between neovim versions when installing as a NixOS module

### DISCUSSION THREAD

**Discussion by michael-koller-91** [nix-community/kickstart-nix.nvim#16] [Q&A]
Thanks for this project!

When I follow the installation instructions and do a nixos-rebuild, I can run `nvim` to start neovim. However, no plugins seem to be installed. By contrast, if I run `nix run .#nvim` in the kickstart-nix.nvim folder, neovim starts with all plugins.
Is `nix run` what I should use to start neovim or is `nvim` supposed to start neovim with all plugins enabled and I did something wrong during the installation?

Thank you very much.

**Comment by mrcjkb** [comment]
Hey 👋 

Do you have any other neovim related configurations in your NixOS config? 
I sounds like something is leading to neovim being added to your systemPackages. If it appears in your PATH before the kickstart-nix package, `nvim` will invoke that one. You can check this by running `realpath "$(which nvim)"` and by comparing the nix store path to the one produced by the kickstart-nix build.

**Reply by michael-koller-91** [reply]
Thanks for your help. `nix run .#nvim` leads to neovim 0.9.5. The only thing related to neovim in systemPackages is `nvim-pkg`. If I do a nixos-rebuild with `nvim-pkg` in systemPackages, it says something about neovim 0.9.4 and if I run `nvim` it leads to neovim 0.9.4. Do you happen to know what the issue might be?

**Reply by mrcjkb** [reply]
Here are some things you can check: 

- See if there's a `~/.nix-profile/bin/nvim` (left over from a previous `nix-env -i` or `nix profile install`).
- Check you `PATH` environment variable. 
- Check your NixOS/home-manager modules for anything neovim related.

Can you run `nvim` if you don't have `nvim-pkg` in your `systemPackages`?

**Reply by michael-koller-91** [reply]
I appreciate your help, thank you.

- There is no `nvim` in `~/.nix-profile/bin/nvim`.
- If I check the `PATH`, there is only `nvim` in `/run/current-system/sw/bin` which is also what `which nvim` gives.
- I never used home-manager for anything neovim related.
- Without `nvim-pkg` I cannot run `nvim`. It says `nvim: command not found`.

**Reply by mrcjkb** [reply]
:thinking: that's very strange.
Do you have a link to your NixOS config?

Normally, it should be enough to add the overlay and then add the package to your systemPackages.
You can also try using the flake output directtly, e.g. 

```nix
environment.systemPackages = [
  inputs.kickstart-nix.packages.${pkgs.system}.nvim-pkg
];
```

**Reply by michael-koller-91** [reply]
My flake.nix and configuration.nix can be found here:
https://github.com/michael-koller-91/.dotfiles/tree/main
I must be doing something wrong...

**Reply by mrcjkb** [reply]
Everything looks fine to me.
🤔 I'll have to try and remodel how you have it set up and see if I can reproduce it...

**Reply by mrcjkb** [reply]
Ah, I can see what's causing the problem in your config.

You have `nvim-kickstart.inputs.nixpkgs.follows = "nixpkgs";` in your flake, which is `nixos-23.11`.

This flake uses `nixpkgs.wrapNeovimUnstable`, which (as the name says) is unstable, and changes its signature.
For example, the attribute `luaRcContent`, which this flake uses to pass the init.lua, does not exist in nixos-23.11. That explains why no plugins are being loaded for you.

**Reply by michael-koller-91** [reply]
Thank you for your investigation! I'll try to find out how to fix my problem.

**Reply by mrcjkb** [reply]
Removing `nvim-kickstart.inputs.nixpkgs.follows = "nixpkgs"; ` from your flake.nix should fix it.

**Reply by michael-koller-91** [reply]
I was hoping that too but I still end up with version 0.9.4. But I really don't want to bother you anymore 😅. You have helped a lot.

**Reply by mrcjkb** [reply]
:thinking: that's strange. You could try adding nixpkgs-unstable as an input and changing it to `nvim-kickstart.inputs.nixpkgs.follows = "nixpkgs-unstable";`.

Or, maybe the input is still pinned to nixpkgs-release-23.11 in your flake.lock?

**Reply by michael-koller-91** [reply]
I did the following:
- delete all old profiles
- `nix-garbage-collect -d`
- reboot
- `ls /nix/store | grep -e "nvim" -e "neovim"` shows no results
- add follows unstable to flake.nix
- delete flake.lock
- `nix flake update`
- `sudo nixos-rebuild switch --flake .`
But it still builds 0.9.4 :(

**Reply by mrcjkb** [reply]
Not a real solution, but does it work if you use `builtins.getFlake` (instead of fetching it via the flake inputs)?
I hate to say it, but I'm starting to run out of ideas :disappointed:

**Reply by michael-koller-91** [reply]
I think I need to give up :( Thank you for all your help!

**Reply by mrcjkb** [reply]
Sorry I couldn't be of more help :disappointed:

**Reply by mrcjkb** [reply]
I've just pushed a fix that should solve this issue for the overlay (as long as you don't set `inputs.nixpkgs.follows` when importing the flake).
